### DISCUSSION TITLE
Usar a estrutura do Next.js para construção de um API

### DISCUSSION THREAD

**Discussion by tcelestino** [frontendbr/forum#1989] [Discussão]
Ultimamente percebo que muita gente descobriu o Next.js para desenvolvimento de aplicações em React.js, utilizando principalmente para entregar solução frontend, porém o desenvolvimento de endpoints que podem ser usados internamente na aplicação ou fora dela, me veio a dúvida se não seria possível usar o Next.js apenas para o desenvolvimento de um API.

Alguém já pensou ou já desenvolveu alguma coisa parecida usando o Next.js, sem a necessidade de usar o frontend? 
Se sim, quais foram os pros e contras de usar a arquitetura do framework para isso?

**Comment by victorferreira** [comment]
É possível, mas me faz questionar: se você só quer usar os endpoints, será que precisa mesmo do Next? Me parece que uma aplicação com o Express seria o suficiente nesse caso.

**Comment by wilsonneto-swe** [comment]
Concordo com o @victorferreira, seria algo que você consegue, mas a ferramenta foi feita com foco em frontend, e os endpoints ali são meio que um segundo plano. Seria bem mais fácil e usando as ferramentas feitas para isso se usar um strapi se quer algo já adiantado e mais abstraído ou ide express, koa ou nest... 

Forte abraço e sucesso!

**Comment by ninetails** [comment]
Se for utilizar o metaframework Next.js para isso, eu não recomendo. Para subir um frontend o Next.js te dá mais abstrações e bibliotecas para te ajudar a produzir o teu site utilizando React, você muito provavelmente não vai precisar desta abstração toda para fazer sua API de backend. A única utilização desta abstração toda que vejo caso utilizada no backend seria se você gerasse saídas de API construídas a partir de componentes renderizados no servidor, isso é muito específico e não é o caso de quase ninguém.

Dado isso, faça tuas APIs o mais simples possível e necessitando do menor código de terceiro possível pois assim você não afetará a performance do teu servidor carregando bibliotecas de forma desnecessária já que não serão utilizadas.

-----

Caso esteja falando sobre utilizar a **estrutura** do Next.js, nada impede caso queira utilizar, mas talvez você possa ter problemas em relação a fazer todo o setup e overengineering pra sua aplicação ficar com uma estrutura semelhante de arquivos.

OU você pode procurar por algum **framework** backend que já tenha estruturas definidas e ver se te satisfaz. Pois o express é praticamente uma biblioteca, não um framework, ele não te obriga a fazer em cima de nenhuma estrutura de arquivos etc. Talvez exista algum framework especializado pra APIs que te dê uma estrutura de pastas bem semelhante já.
